### ISSUE TITLE
Autostart problem

### ISSUE THREAD

**Issue by duquediazn** [Gibtnix/MSIKLM#56]
Hi,
I would like to thank you for the excellent work on MSIKLM. I have tested running msiklm manually, and it works correctly. However, I encountered a problem when setting up autostart on my system. When I tried to configure the autostart, the system failed to boot correctly and became unresponsive.

I tried: 
`./autostart.sh green `

The /etc/udev/rules.d/90-msiklm.rules file generated contains:
`ACTION=="add", ATTRS{idVendor}=="1770", ATTRS{idProduct}=="ff00", RUN+="/usr/local/bin/msiklm 'green'"
`
This causes msiklm to fail with exit code 255 during system startup. 

Here is the relevant part of the error log:
```
jun 09 23:32:35 GS63-7RE systemd-udevd[472]: 0003:1770:FF00.0033::side: Process '/usr/local/bin/msiklm 'green'' failed with exit code 255.
jun 09 23:32:35 GS63-7RE systemd-udevd[469]: 0003:1770:FF00.0033::back: Process '/usr/local/bin/msiklm 'green'' failed with exit code 255.
...
jun 09 23:33:35 GS63-7RE systemd-udevd[455]: 0003:1770:FF00.003B::front: Spawned process '/usr/local/bin/msiklm 'green'' [1735] is taking longer than 59s to complete.
```
I forced a system reboot and entered recovery mode to execute:
`./autostart.sh --disable
`
and was able to start the system again.

My laptop: MSI GS63 7RE Stealth Pro
Operating System: Linux Mint 21.3                   
Kernel: Linux 5.15.0-112-generic

**Comment by AtomicRobotMan0101** [comment]
Can I suggest a read of my post here: https://github.com/Gibtnix/MSIKLM/issues/54 <sup>_It could do with a bit of a tidyup though...._</sup> and consider creating a service via SystemD rather than udev?

(**edit** - Ive found the udev method tends to bash the hell out of the system. It restarts a billion times a second.)

I've been given a few old MSI gaming laptops for a program I run (a charity) and this is one way that Ive crushed the problem.

I'll let @Gibtnix officially reply, obviously.

I felt it necessary to reply as I still receive the archive digests via email (!!! cool!) and I believe that with the number of Windows people fleeing to _Greener Pastures (Linux!)_ this project will get a bit more attention.... people will recycle aging machines as they are perfectly good daily drivers :)

**Comment by duquediazn** [comment]
Thank you so much for your response and for sharing your solution. I followed your instructions to create a systemd service for MSIKLM, and it works perfectly. I no longer experience the issues I had with the udev method. :+1: 

Your post was incredibly helpful, and I appreciate the effort you put into explaining the steps. I ended up making my [own version](https://github.com/duquediazn/MSIKLM/blob/main/autostart.sh) of the autostart script :)
